package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -726,6 +777,635 @@ def __setstate__(self, state):
+        # it exists. If these are True, the writer will check each
+        # Timestep.has_*  value and fill the self._has dictionary accordingly
+        # in _initialize_hdf5_datasets()
+        self._write_positions = kwargs.get('positions', True)
+        self._write_velocities = kwargs.get('velocities', True)
+        self._write_forces = kwargs.get('forces', True)
+        if not any([self._write_positions,
+                    self._write_velocities,
+                    self._write_velocities]):
+            raise ValueError("At least one of positions, velocities, or "
+                             "forces must be set to ``True``.")
+
+        self._new_units = {'time': kwargs.pop('timeunit', None),
+                           'length': kwargs.pop('lengthunit', None),
+                           'velocity': kwargs.pop('velocityunit', None),
+                           'force': kwargs.pop('forceunit', None)}
+
+        # Pull out various keywords to store metadata in 'h5md' group
+        self.author = kwargs.pop('author', 'N/A')
+        self.author_email = kwargs.pop('author_email', None)
+        self.creator = kwargs.pop('creator', 'MDAnalysis')
+        self.creator_version = kwargs.pop('creator_version', mda.__version__)
+
+    def _write_next_frame(self, ag):
+        """Write information associated with ``ag`` at current frame
+        into trajectory
+
+        Parameters
+        ----------
+        ag : AtomGroup or Universe
+
+        """
+        try:
+            # Atomgroup?
+            ts = ag.ts
+        except AttributeError:
+            try:
+                # Universe?
+                ts = ag.trajectory.ts
+            except AttributeError:
+                errmsg = "Input obj is neither an AtomGroup or Universe"
+                raise TypeError(errmsg) from None
+
+        if ts.n_atoms != self.n_atoms:
+            raise IOError("H5MDWriter: Timestep does not have"
+                          " the correct number of atoms")
+
+        # This should only be called once when first timestep is read.
+        if self.h5md_file is None:
+            self._determine_units(ag)
+            self._open_file()
+            self._initialize_hdf5_datasets(ts)
+
+        return self._write_next_timestep(ts)
+
+    def _determine_units(self, ag):
+        """determine which units the file will be written with
+
+        By default, it fills the :attr:`self.units` dictionary by copying the
+        units dictionary of the parent file. Because H5MD files have no
+        standard unit restrictions, users may pass a kwarg in ``(timeunit,
+        lengthunit, velocityunit, forceunit)`` to the writer so long as
+        MDAnalysis has a conversion factor for it (:exc:`ValueError` raised if
+        it does not). These custom unit arguments must be in
+        `MDAnalysis notation`_. If custom units are supplied from the user,
+        :attr`self.units[unit]` is replaced with the corresponding
+        ``unit`` argument.
```

Comment:
This text will not be visible in the docs even though it contains useful information. Please summarize it as part of the section on units in the writer notes.